guides/STAKING-VALIDATOR.md

```diff
@@ -0,0 +1,308 @@
+# Setting Up a Solana Validator for Staking with Pulumi and TypeScript
+
+## Introduction
+
+This guide will walk you through enhancing your existing Solana validator setup with additional functionality using Pulumi and TypeScript. Specifically, we'll focus on funding your validator identity and creating vote and stake accounts.
+
+## Prerequisites
+
+- Existing Pulumi TypeScript project for Solana validator setup
+- Basic understanding of Solana validators
+- Familiarity with TypeScript and Pulumi
+- Necessary access to fund accounts
+
+## Process Overview
+
+We'll add the following components to your existing setup:
+
+1. Create a treasury keypair for funding
+2. Fund the validator identity
+3. Create a vote account
+4. Create a stake account
+5. Export public keys as Pulumi outputs
+
+## Existing Setup
+
+Your current setup already includes:
+- AWS infrastructure (EC2 instance, SSH key)
+- Network configuration
+- Validator and vote account keypairs
+- Validator tuning configuration
+- Agave validator installation
+
+## Implementation
+
+Let's enhance your existing `index.ts` file with the additional components. Here's where to add the new code:
+
+### 1. Add a Treasury Keypair
+
+After your validator and vote account keypair definitions, add:
+
+```typescript
+// Add a treasury key for funding validator operations
+const treasuryKey = new svmkit.KeyPair("treasury-key");
+
+// Export the treasury public key so you can fund it externally
+export const treasuryPublicKey = treasuryKey.publicKey;
+```
+
+### 2. Fund the Validator Identity
+
+Using the `validatorPublicKey` fund it with sol as well the `treasuryPublicKey`. Both need to be funded before proceed (for `devnet` test it's recommended to fund 0.1 sol the `treasuryPublicKey` and 0.3 the `validatorPublicKey`).
+https://solana.com/developers/guides/getstarted/solana-token-airdrop-and-faucets
```

Comment:
This is good to say pause and fund but to combine the lower section just cover when to pulumi up.